mission_6/sprint-mission-6.sql

```diff
@@ -0,0 +1,146 @@
+
+-- 4. `orders` 테이블에서 각 날짜별 총 주문 건수를 `order_count` 라는 이름으로 계산하고, 하루 총 주문 건수가 80건 이상인 날짜만 조회한 뒤, 주문 건수가 많은 순서대로 정렬하세요.
+SELECT * FROM (SELECT date, COUNT(*) AS order_count FROM orders GROUP BY date) WHERE order_count >= 80 ORDER BY order_count DESC;
+
+-- 5. `order_details` 테이블에서 피자별(`pizza_id`) 총 주문 수량이 10개 이상인 피자만 조회하고, 총 주문 수량 기준으로 내림차순 정렬하세요.
+SELECT * FROM (SELECT pizza_id, sum(quantity) AS total_sum FROM order_details GROUP BY pizza_id) WHERE total_sum >= 10 ORDER BY total_sum DESC;
+
+-- 6. `order_details` 테이블에서 피자별 총 수익을 `total_revenue` 라는 이름으로 구하세요. (수익 = `quantity * price`)
+SELECT od.pizza_id, SUM(od.quantity * pz.price) AS total_revenue FROM order_details AS od JOIN pizzas AS pz ON od.pizza_id = pz.id GROUP BY od.pizza_id ORDER BY od.pizza_id;
+
+-- 7. 날짜별로 피자 주문 건수(`order_count`)와 총 주문 수량(`total_quantity`)을 구하세요.
+SELECT o.date, COUNT(DISTINCT o.id) AS order_count, SUM(d.quantity) AS total_quantity FROM orders AS o JOIN order_details AS d ON o.id = d.order_id GROUP BY o.date ORDER BY o.date DESC;
+
+-- # 고급 문제
+
+/*
+    1. 피자별(`pizzas.id` 기준) 판매 수량 순위에서 피자별 판매 수량 상위에 드는 베스트 피자를 10개를 조회해 주세요. `pizzas`의 모든 컬럼을 조회하면서 각 피자에 해당하는 판매량을 `total_quantity`라는 이름으로 함께 조회합니다.
+        
+        출력 예시:
+
+        ```sql
+            big_meat_s    | big_meat    | S    |    12 |           1914
+            thai_ckn_l    | thai_ckn    | L    | 20.75 |           1410
+            five_cheese_l | five_cheese | L    |  18.5 |           1409
+            four_cheese_l | four_cheese | L    | 17.95 |           1316
+            classic_dlx_m | classic_dlx | M    |    16 |           1181
+            spicy_ital_l  | spicy_ital  | L    | 20.75 |           1109
+            hawaiian_s    | hawaiian    | S    |  10.5 |           1020
+            southw_ckn_l  | southw_ckn  | L    | 20.75 |           1016
+            bbq_ckn_l     | bbq_ckn     | L    | 20.75 |            992
+            bbq_ckn_m     | bbq_ckn     | M    | 16.75 |            956
+        ```
+*/
+SELECT p.*, SUM(d.quantity) FROM pizzas AS p JOIN order_details AS d ON p.id = d.pizza_id GROUP BY p.id ORDER BY SUM(d.quantity) DESC;
+
+
+/*
+    2. `orders` 테이블에서 2025년 3월의 일별 주문 수량을 `total_orders`라는 이름으로, 일별 총 주문 금액을 `total_amount`라는 이름으로 포함해서 조회하세요.
+        
+        출력 예시:
+        
+        ```sql
+        2025-03-01 |           49 | 1598.5500011444092
+        2025-03-02 |           58 |  2379.050001144409
+        2025-03-03 |           53 | 2287.8999996185303
+        2025-03-04 |           59 |  2444.300001144409
+        2025-03-05 |           64 |  2350.650005340576
+        ```
+*/
+SELECT o.date, COUNT(o.id) AS total_orders, SUM(t.sum) AS total_amount FROM orders AS o JOIN (SELECT o.order_id, SUM(o.quantity * p.price) FROM order_details AS o JOIN pizzas AS p ON o.pizza_id = p.id GROUP BY o.order_id) AS t ON o.id = t.order_id WHERE o.date BETWEEN '2025-03-01' AND '2025-03-31' GROUP BY o.date ORDER BY o.date ASC;
```

Comment:
1. 서브 쿼리의 집계 컬럼에 별칭이 없어서, 바깥에서 참조(`t.sum`)가 안됩니다.
2. 가독성을 위해 별칭을 분리해서 사용해주세요.
```suggestion
SELECT o.date,
       COUNT(o.id)                       AS total_orders,
       SUM(t.order_amount)               AS total_amount
FROM orders AS o
JOIN (
  SELECT od.order_id,
         SUM(od.quantity * p.price)      AS order_amount
  FROM order_details AS od
  JOIN pizzas AS p ON od.pizza_id = p.id
  GROUP BY od.order_id
) AS t
  ON o.id = t.order_id
WHERE o.date BETWEEN '2025-03-01' AND '2025-03-31'
GROUP BY o.date
ORDER BY o.date ASC;
```